test/Governance.ts

```diff
@@ -1324,6 +1324,65 @@ describe("Governance test", function () {
     });
 });
 
+describe.only("Proposals test", function () {
+    beforeEach(async function (){
+        Object.assign(this, await loadFixture(governanceFixture));
+    });
+    it("SoftwareUpgradeProposal", async function () {
+        const upgradableProxy: UpgradableCounterContract = await upgrades.deployProxy(
+            await ethers.getContractFactory('UpgradableCounterContract'),
+            [await this.gov.getAddress()],
+            {kind: 'uups'}
+        );
+        await this.verifier.addTemplate(
+            3,
+            "templ",
+            ethers.ZeroAddress,
+            DelegateCallType,
+            ethers.parseEther("0.5"),
+            ethers.parseEther("0.6"),
+            [0, 1, 2, 3, 4],
+            1,
+            2,
+            0,
+            100
+        );
+
+        // Deploy new implementation
+        const newImplContract = await ethers.deployContract('UpgradableCounterContract');
+        await newImplContract.initialize(await this.gov.getAddress());
+
+        // Create the proposal with the current implementation (proxy) and the new implementation as the proposed one
+        const proposalContract = await ethers.deployContract(
+            "SoftwareUpgradeProposal",
+            [
+                "upgrade",
+                "upgrade-descr",
+                ethers.parseEther("0.5"),
+                ethers.parseEther("0.6"),
+                0,
+                1,
+                2,
+                upgradableProxy,
+                newImplContract,
+                ethers.ZeroAddress,
+                "0x"
```

Comment:
add comments explaining the comments meaning, avoid not-self-explaning literals:
```
  ethers.parseEther("0.5"), // minVotes
  ethers.parseEther("0.6"), // minAgreement
  0, // start time
  1, // min end time
  2, // max end time
```
(even better if named constants can be introduced, but not necessary)